### DISCUSSION TITLE
Company password saves?

### DISCUSSION THREAD

**Discussion by Shituation** [OpenTTD/OpenTTD#10999] [Polls]
Here we are, in 2023 and we, game admins can't reboot our servers because we lose all the company passwords from our users including our own company if we have it. We cannot schedule reboots, we cannot apply patches, cannot relax if we have an electric service blackout.

Devs refuse to implement this feature saying people can be hacked since 2005 like if knowing a random password could be used for something. And in the remote possibility someone can do something with them, its user's fault, same if I were using my bank account password in any game, service. We can burn our houses down by using the oven, what do we do? ban them?🙃

https://github.com/OpenTTD/OpenTTD/issues/8303

Every single post about this gets discarded. A guy from the forums made a patch back in 2007, they cancelled him and didn't add the patch to the code. The patch made a file with company passwords separated from the savegame.
https://www.tt-forums.net/viewtopic.php?t=34997

**Comment by 2TallTyler** [comment]
The patch in the forum topic was not rejected. It looks unfinished from what I read on Flyspray. It can't be simply added now, as the code is old enough to get a driver's license in some countries, and would need some updating. 😉 

#8303 was not rejected, nor was there a refusal to add the feature: it was closed because it is not a bug. It was also explained that there is a planned solution (persistent clients) to the implicit feature request in the future.

If you want to add this feature (say, from the patch as a separate file) and submit a Pull Request, I'm sure it would be fairly considered. But submitting a poll bashing "the developers" isn't going to make anyone do what you want. OpenTTD is not a democracy where the players choose how volunteer developers spend their free time. If you want something, put in the work to contribute it yourself.

My mom taught me that if you don't have anything nice to say, don't say anything. We have blocked people from posting here after repeated rude messages, and will do it again if needed. We don't get paid to listen to complaints. So please be nice. 🙂

**Reply by Shituation** [reply]
What is rude? to dissent? if you are in power others can't challenge you?   ...is that democracy? 
I can't put up a poll to see what people wants? 
Is it not true that every single time someone wanted this feature, it was rejected? 

I'm not being rude here as far as I can see.

**Reply by embeddedt** [reply]
A feature not being implemented is not the same as rejection; there is limited development effort in a volunteer project.

**Reply by masterofobzene** [reply]
"My mom taught me that if you don't have anything nice to say, don't say anything. "

Wow... that is so dangerous...

**Reply by agentw4b** [reply]
I ran a multiplayer server for several years and this was a big problem for years, not being able to restart the server without losing company passwords. Most of the time I ended up having to start a whole new game. And nothing has changed since then.

**Reply by MuxyDuGoulp** [reply]
I ran also multiplayer serverS for a long time, and this has never been a problem.
Of course server was patched to save the companies passwords as they were sent (encrypted).
Also have a patch in order to have mandatory passwords.
of course savegame files is not a place for passwords.
anyway, problem solved.

**Reply by agentw4b** [reply]
Putting some unspecified patches into the base game that don't exist in the official game isn't a solution, it's part of the problem....

**Reply by MuxyDuGoulp** [reply]
When something is not available at the time you need it and dont want to deal with dev cause you dont know when and if it will be available in prod, then you do it yourself. Its not a problem at all.

**Comment by frenchtrainspotter** [comment]
The problem with making a poll here is that, sadly, most people able to answer to this poll know that they shouldn’t use a social/bank password… And everything else that has to be said was already said.

**Comment by rubidium42** [comment]
Company passwords have been removed in master (and as such 15.0). Instead there now is a password less mechanism that does save state. So reloading will just retain the information who is allowed to join what company.

Next to that the password authentication of the game has been changed so no password needs to be sent to the server, as there are other methods to validate knowledge of a password without actually sending the password.

So, people using their bank password for the server is not a big problem any more as the password is not sent over the network; it is still bad practice, but OpenTTD itself does not allow leaking it via unencrypted channels. Next to that people cannot use their bank password for companies, and the authentication state of people is retained so server restarts should not be a problem any more.

Also the rcon password is not sent without encryption any more, although I would advice to add the appropriate client keys to the server's configuration so that can be performed password less. Finally there is a PR to do the same for the admin connection.
